R/process_sc_all_home_care.R

```diff
@@ -0,0 +1,186 @@
+#' Process the all home care extract
+#'
+#' @description This will read and process the
+#' all home care extract, it will return the final data
+#' but also write this out as a rds.
+#'
+#' @param data The extract to process
+#' @param sc_demographics The sc demographics lookup. Default set to NULL as
+#' we can pass this through data in the environment.
+#' @param write_to_disk (optional) Should the data be written to disk default is
+#' `TRUE` i.e. write the data to disk.
+#'
+#' @return the final data as a [tibble][tibble::tibble-package].
+#' @family process extracts
+#'
+process_sc_all_home_care <- function(data, sc_demographics = NULL, write_to_disk = TRUE) {
+  # Match on demographic data ---------------------------------------
+  if (is.null(sc_demographics)) {
+    # read in demographic data
+    sc_demog <- readr::read_rds(get_sc_demog_lookup_path())
+  }
+
+  matched_hc_data <- data %>%
+    dplyr::left_join(sc_demog, by = c("sending_location", "social_care_id"))
+
+
+  # Data Cleaning ---------------------------------------
+
+  # period start and end dates
+  period_dates <- matched_hc_data %>%
+    dplyr::distinct(.data$period) %>%
+    dplyr::mutate(
+      record_date = end_fy_quarter(.data$period),
+      qtr_start = start_fy_quarter(.data$period)
+    )
+
+  home_care_clean <- matched_hc_data %>%
+    # set reablement values == 9 to NA
+    dplyr::mutate(reablement = dplyr::na_if(.data$reablement, "9")) %>%
+    # fix NA hc_service
+    dplyr::mutate(hc_service = tidyr::replace_na(.data$hc_service, "0")) %>%
+    # join with dates
+    dplyr::left_join(.data$period_dates, by = c("period")) %>%
+    # Replace missing start dates with the start of the quarter
+    dplyr::mutate(hc_service_start_date = dplyr::if_else(is.na(.data$hc_service_start_date), .data$qtr_start, .data$hc_service_start_date)) %>%
+    # Replace really early start dates with start of the quarter
+    dplyr::mutate(hc_service_end_date = dplyr::if_else(.data$hc_service_start_date < as.Date("1989-01-01"), .data$qtr_start, .data$hc_service_start_date)) %>%
+    # when multiple social_care_id from sending_location for single CHI
+    # replace social_care_id with latest
+    dplyr::group_by(.data$sending_location, .data$social_care_id) %>%
+    dplyr::mutate(latest_sc_id = dplyr::last(.data$social_care_id)) %>%
+    # count changed social_care_id
+    dplyr::mutate(
+      changed_sc_id = dplyr::if_else(!is.na(.data$chi) & .data$social_care_id != .data$latest_sc_id, 1, 0),
+      social_care_id = dplyr::if_else(!is.na(.data$chi) & .data$social_care_id != .data$latest_sc_id,
+        .data$latest_sc_id, .data$social_care_id
+      )
+    ) %>%
+    dplyr::ungroup() %>%
```

Comment:
Can this block of code be extracted as its own function to be used elsewhere, please?

```R
replace_sc_id_with_latest <- function(data, ..?) {
# Check for required variables

# Filter to CHIs with multiple sc_id per sending loc

# Sort (by period?) to find the latest sc_id for each CHI / sending_loc

# match that back to the data overwriting the older one.

# Return the data (no new variables added, just overwrite the sc_id)

}
```

Reply:
Already an issue! #360

Reply:
PR open #440